stdlib/REPL/src/docview.jl

```diff
@@ -728,8 +754,18 @@ function fuzzyscore(needle::AbstractString, haystack::AbstractString)
     1 - (string_distance(needle, lena, haystack, lenb) / max(lena, lenb))
 end
 
-function fuzzysort(search::String, candidates::Vector{String})
-    scores = map(cand -> fuzzyscore(search, cand), candidates)
+function fuzzyscore(needle::AbstractString, haystack::AccessibleBinding)
+    score = fuzzyscore(needle, haystack.name)
+    haystack.source === nothing && return score
+    # Apply a "penalty" of half an edit if the comparator binding is public but not
+    # exported so that exported/local names that exactly match the search query are
+    # listed first
+    penalty = 1 / (2 * max(length(needle), length(haystack.name)))
+    return max(score - penalty, 0)
+end
+
+function fuzzysort(search::String, candidates::Vector{AccessibleBinding})
+    scores = map(cand -> fuzzyscore(search, cand.name), candidates)
```

Comment:
I think I found it: the sort order only considers the name here, so it never gets penalized for a source, making their order unpredictable:
```suggestion
    scores = map(cand -> fuzzyscore(search, cand), candidates)
```